src/main/java/com/agilethought/atsceapi/validator/HistoryValidator.java

```diff
@@ -0,0 +1,120 @@
+package com.agilethought.atsceapi.validator;
+
+import static com.agilethought.atsceapi.exception.ErrorMessage.BAD_REQUEST_MESSAGE;
+import static com.agilethought.atsceapi.exception.ErrorMessage.BAD_REQUEST_MESSAGE_REQUIRED_NUMERIC;
+import static com.agilethought.atsceapi.exception.ErrorMessage.BAD_REQUEST_MESSAGE_HISTORY_TYPE;
+import static com.agilethought.atsceapi.exception.ErrorMessage.BAD_REQUEST_MESSAGE_STATUS;
+import static com.agilethought.atsceapi.exception.ErrorMessage.BAD_REQUEST_MESSAGE_USER_ID;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.agilethought.atsceapi.dto.NewHistoryRequest;
+import com.agilethought.atsceapi.exception.BadRequestException;
+
+@Service
+public class HistoryValidator implements Validator<NewHistoryRequest> {
+    @Autowired
+	private FixedExpensesValidator fixedExpensesValidator;
+    
+	@Override
+	public void validate(NewHistoryRequest historyRequest) {
+		validateHistoryTypeField(historyRequest.getType());
+		validateUserId(historyRequest.getUser_id());
+		validateTotalHours(historyRequest.getTotalHours());
+		validateTotalDays(historyRequest.getTotalDays());
+		validateCostDay(historyRequest.getCostDay());
+		validateCostHour(historyRequest.getCostHour());
+		validateProjectCost(historyRequest.getProjectCost());
+		validateTaxIVA(historyRequest.getTaxIVA());
+		validateTaxISR_r(historyRequest.getTaxISR_r());
+		validateTaxIVA_r(historyRequest.getTaxIVA_r());
+		validateTotal(historyRequest.getTotal());
+		validateRevenue(historyRequest.getRevenue());
+		validateHistoryStatus(historyRequest.getStatus());
+		fixedExpensesValidator.validate(historyRequest.getFixedExpenses());
+	}
+
+	private void validateHistoryTypeField(Integer type) {
+		if (type == null || type < 1 || type > 4) {
+			throw new BadRequestException(BAD_REQUEST_MESSAGE + "*TYPE* " + BAD_REQUEST_MESSAGE_HISTORY_TYPE);
+		}
+	}
+
+	private void validateUserId(String user_id) {
+		if (StringUtils.isBlank(user_id)) {
+			throw new BadRequestException(BAD_REQUEST_MESSAGE + "*USER_ID* " + BAD_REQUEST_MESSAGE_USER_ID);
+		}
+	}
+
+	private void validateTotalHours(Integer totalHours) {
+		if (totalHours == null || totalHours < 0) {

```

Comment:
`if (totalHours == null || totalHours < 0)` we are repeating a lot of time this validation. Maybe it's a good idea have a util method to generify ` utilMethod(intVar) -> return intVar== null || intVar;`